prov/efa/src/efa_mr.c

```diff
@@ -811,6 +813,25 @@ static int efa_mr_reg_impl(struct efa_mr *efa_mr, uint64_t flags, const void *at
 	if (ret)
 		return ret;
 
+	device_support_rdma_read = efa_mr->domain->device->device_caps & EFADV_DEVICE_ATTR_CAPS_RDMA_READ;
+#if HAVE_CAPS_RDMA_WRITE
+	device_support_rdma_write = efa_mr->domain->device->device_caps & EFADV_DEVICE_ATTR_CAPS_RDMA_WRITE;
+#endif
+	/* For efa-direct, fail registration if RDMA operations are requested 
+	 * but hardware doesn't support them */
+	if (!strcmp(efa_mr->domain->util_domain.fabric->name, EFA_DIRECT_FABRIC_NAME)) {
+		if ((mr_attr.access & (FI_READ | FI_REMOTE_READ)) && !device_support_rdma_read) {
+			EFA_WARN(FI_LOG_MR,
+				 "FI_READ requested but hardware does not support RDMA read operations\n");
```

Comment:
nit: log entry says `FI_READ` but it can also be printed when only `FI_REMOTE_READ` is requested - which would be confusing to the user